packages/esm-messages-app/src/messages/messages-chart-dashboard.component.tsx

```diff
@@ -0,0 +1,195 @@
+      });
+    },
+    [messagesTemplates, mutateTemplates],
+  );
+
+  if (isLoadingTemplates) return <DataTableSkeleton role="progressbar" compact={isDesktop} zebra />;
+  if (error) return <ErrorState error={error} headerTitle={headerTitle} />;
+  if (messagesTemplates?.length) {
+    return (
+      <div className={styles.widgetCard}>
+        <CardHeader title={headerTitle}>
+          <span>{isValidatingTemplates ? <InlineLoading /> : null}</span>
+          <div className={styles.rightMostFlexContainer}>
+            <Button
+              kind="ghost"
+              renderIcon={(props) => <Add size={16} {...props} />}
+              iconDescription="Add messages"
+              onClick={() => {}}
+            >
+              {t('add', 'Add')}
+            </Button>
+          </div>
+        </CardHeader>
+        <DataTable
+          rows={tableRows}
+          headers={headers}
+          isSortable
+          size={isTablet ? 'lg' : 'sm'}
+          useZebraStyles
+          overflowMenuOnHover={isDesktop}
+        >
+          {({ rows, headers, getHeaderProps, getTableProps, getRowProps }) => (
+            <>
+              <TableContainer>
+                <Table aria-label="messages summary" {...getTableProps()}>
+                  <TableHead>
+                    <TableRow>
+                      {headers.map((header) => (
+                        <TableHeader
+                          className={classNames(styles.productiveHeading01, styles.text02)}
+                          {...getHeaderProps({
+                            header,
+                            isSortable: header.isSortable,
+                          })}
+                        >
+                          {header.header?.content ?? header.header}
+                        </TableHeader>
+                      ))}
+                      <TableHeader />
+                    </TableRow>
+                  </TableHead>
+                  <TableBody>
+                    {rows.map((row) => (
+                      <TableRow key={row.id} {...getRowProps({ row })}>
+                        {row.cells.map((cell) => (
+                          <TableCell key={cell.id}>{cell.value?.content ?? cell.value}</TableCell>
+                        ))}
+                        <TableCell className="cds--table-column-menu">
+                          <Layer className={styles.layer}>
+                            <OverflowMenu
+                              aria-label="Edit message type"
+                              size={isTablet ? 'lg' : 'sm'}
+                              flipped
+                              align="left"
+                            >
+                              <OverflowMenuItem
+                                className={styles.menuItem}
+                                id="editMessageType"
+                                onClick={() => launchEditMessagesForm(row.id)}
+                                itemText={t('edit', 'Edit')}
```

Comment:
Many common strings, like `edit` and `save` are available via the `getCoreTranslation()` API call. We prefer to use this where possible as it saves our translators from having to re-translate the same string for every new app.